### PR TITLE
feat: add retry logic to libicu download

### PR DESCRIPTION
**Pull request by jkruse14**
## Background
---
There seems to be a race condition some where with the user-data script and the EC2 starting up and locking RPM. This issue is seen elsewhere, not necessarily with this repo's user-data, see the following:

[Amazon Linux 2023 - issue with installing packages with cloud-init](https://repost.aws/questions/QU_tj7NQl6ReKoG53zzEqYOw/amazon-linux-2023-issue-with-installing-packages-with-cloud-init)

[dnf/yum both fails while being executed on instance bootstrap on Amazon Linux 2023](https://repost.aws/questions/QUgNz4VGCFSC2TYekM-6GiDQ/dnf-yum-both-fails-while-being-executed-on-instance-bootstrap-on-amazon-linux-2023)

Also, #3741

## Changes Made
---
Added a loop to retry if the rpm lock file is found which sleeps for 5 seconds then retries again with a total of 5 iterations.

## Testing
---
Spun up an instance and ran the script directly (using sudo the second time because this was not run via user-data)
image

### CODE REVIEW

**Comment by jkruse14** [comment]
@npalm - any chance we could get eyes on this?

**Review by npalm — COMMENTED** [review]
Thanks, looks good. Need to test asap

**Comment by jkruse14** [comment]
> Thanks, looks good. Need to test asap

@npalm  - Anything you need from me for testing?

File: modules/runners/templates/install-runner.sh [hunk 1]
```diff
@@ -47,7 +47,20 @@ echo OS: $os_id
 
 # Install libicu on non-ubuntu
 if [[ ! "$os_id" =~ ^ubuntu.* ]]; then
-  dnf install -y libicu
+  max_attempts=5
+  attempt_count=1
```
Comment (npalm) [inline modules/runners/templates/install-runner.sh:51]:
```suggestion
  attempt_count=0
```

**Review by npalm — CHANGES_REQUESTED** [review]
Looks good, a small suggestion

**Review by npalm — APPROVED** [review]

**Pull request merged**